### ISSUE TITLE
What happens if client didn't disconnect properly?

### ISSUE THREAD

**Issue by AmirOsman2020** [centrifugal/centrifugo#548]
Hello everyone, I'm building an android chat app using the java SDK but I'm wondering, if someone closed the app without properly disconnecting from Centrifugo server, will the connection still be opened?

**Comment by FZambia** [comment]
Hello, this question is not for Centrifugo issue tracker.

See https://github.com/centrifugal/centrifuge-java#usage-in-background. The connection may be still open for some time, but it may be closed at some point due to OS strategy. So there are no guarantees client will properly work in the background. When app is in the background applications usually use push notifications instead of real-time messaging.

**Comment by AmirOsman2020** [comment]
No, I'm not talking about the client am talking about the server, if the client didn't disconnect properly will the connection still be open on the server? I'm just concerned about how Centrifugo handles improper disconnections.

**Comment by FZambia** [comment]
> No, I'm not talking about the client am talking about the server, if the client didn't disconnect properly will the connection still be open on the server?

Got it, yes - Centrifugo handles this, connection may be closed whether properly (like WebSocket close handshake), or due to server-to-client ping-pong mechanism which allows finding stale connections within ~25 seconds interval.

**Comment by AmirOsman2020** [comment]
25 seconds interval seems like a long time, can I reduce that to 15 seconds?

**Comment by FZambia** [comment]
Not at the moment, opened an issue to tune ping timings of a server - https://github.com/centrifugal/centrifugo/issues/549
